introduction.md

```diff
@@ -0,0 +1,435 @@
+places containers onto members of the cluster and then can instruct
+the agents to create and manage network connections between
+containers.
+
+By itself the host clustering layer does not make intelligent resource
+allocation decisions. Without guidance it merely places each new
+container on some available host.
+
+#### Orchestration Layer 3: Network Resources and Publication
+
+All of the focus up until now has been on activities within the
+container host cluster: Creating and connecting containers with each
+other. The third layer of the orchestration system looks outside the
+cluster to bring resources in for the containers to use.
+
+The most common external resources are network storage and inbound
+connectivity. The first is a pull operation, bringing known resources
+to the container hosts and making them available to the container
+system. The second is a kind of push operation and requires two steps:
+
+1. Establish an inbound access point and a path to the container(s)
+1. Publish the inbound access point
+
+### Resource Management, Access Control and Policy
+
+Properly, resource management is outside the scope of the
+orchestration system. The decisions about resource allocation are made
+by a policy engine such as _Mesos_. However, the orchestration system
+must be able to reflect and report on the resources that are
+available, and it must be able to accept allocation requests which
+respect the decisions of the policy engine. This means it must be able
+to both label resources and allow the policy engine to make requests
+using those labels.
+
+### Summary of Container Management Concepts
+
+[TBD]
+
+## Containers and Orchestration with Kubernetes
+
+[Improve Segue - addressing the frameworks - skipping right to orchestration]
+
+When you want to use software containers to build and run complete
+applications and services, it quickly becomes evident that creating
+single containers on individual hosts will become time consuming and
+unwieldy. Containers run on single hosts. To make services span
+multiple hosts, each host must be prepared with identical remote
+storage and networking resources independent of the container service.
+
+### Container Deployment
+
+_Kubernetes_, an orchestration and clustering layer above the container
+systems provides a means to create individual containers and related
+groups of containers. It provides a way to describe individual
+containers or clusters of containers (_Pods_) which will run together
+and will be assigned to a single container host. It also provides a
+means to create and maintain a specified number of replicas of each
+pod. Replicas will be placed on different nodes.
+
+### Resource Allocation
+
+Kubernetes allocation mechanism is fairly simple. It does not offer
+any way to partition the cluster or to assign pods to sub-sets of the
+cluster for resource or policy management purposes. A Kubernetes
+cluster is flat and uniform. Other than the constraint that 
```

Comment:
i have to go back and look at the context and the narrative flow to see if this needs more, and what.